src/onnx/onnx.cpp

```diff
@@ -1759,18 +1759,28 @@ struct onnx_parser
             this->parse_node(output.name());
         }
 
-        // For now, the last output with a valid name is considered
-        // as the program output, and add an identity instruction at
-        // the program end
+        // Find instructions corresponding to the output
         auto prog_output = graph.output();
-        auto oit         = std::find_if(prog_output.rbegin(), prog_output.rend(), [](auto& node) {
-            return !node.name().empty();
-        });
-
-        if(instructions.count(oit->name()) > 0)
-        {
-            prog.add_instruction(op::identity{}, instructions[oit->name()]);
-        }
+        std::vector<std::string> all_output_names;
+        std::vector<std::string> prog_output_names;
+        std::transform(prog_output.begin(),
+                       prog_output.end(),
+                       std::back_inserter(all_output_names),
```

Comment:
Why do we need to copy this to an extra vector? Why can't the `copy_if` use `prog_output` directly instead?

Reply:
According to our discussion, I will let it unchanged.